sp/src/public/vscript/ivscript.h

```diff
@@ -645,8 +631,21 @@ struct ScriptEnumDesc_t
 // 
 //-----------------------------------------------------------------------------
 
+// forwards T (and T&) if T is neither enum or an unsigned integer
```

Comment:
gcc complained that it can't discern how to create a `ScriptVariant_t` from enums and unsigned integers.
This awkward template ensures it can deal with it properly via an implicit cast to int due to the overload below.

Reply:
You'll still need `std::forward<T>` on this if your expecting to forward it.

Reply:
Indeed, and fix the return-type from `T` to `T&&`.